openedx/core/lib/tests/assertions/events.py

```diff
@@ -6,7 +6,7 @@
 import six
 
 
-def assert_event_matches(expected, actual, tolerate=None):
+def assert_event_matches(expected, actual, tolerate=None, string_payload_key=None):
```

Comment:
not sure why we need these changes.

Reply:
@ziafazal there were two tests that had ` "data": '"GET" :{}, "POST": {}`. While running those tests, those often failed due to the different positioning of `GET` and `POST` in the string. JSON comparison was needed. `assert_event_matches` provides the option to parse JSON data before comparison but it does the parsing only for the `event` field. 
when we changed the event, the `event` field got transformed into the `data` field and thus we need to parse `data` field now. 
This extra parameter is for this purpose.